### DISCUSSION TITLE
搭建公用GPU服务器过程记录 | SHEN's BLOG

### DISCUSSION THREAD

**Discussion by veraposeidon** [veraposeidon/veraposeidon.github.io#92] [General]
https://shenxiaohai.me/2018/06/28/gpu-server/

**Comment by fungtion** [comment]
有几个问题想请教：
1. 在ubuntu16.04下如何配置网桥，按照《ZFS 与 LXC 与 GPU Passthrough，以及贵校超算队集群管理》配置的话，我的服务器无法使用网络。
2. lxd的容器的安装位置能改变吗？
3. 按照您的方法创建的容器，能在另一台电脑上通过ssh访问吗？

**Comment by veraposeidon** [comment]
1. 关于配置网桥这块，我其实不是很懂。我觉得要实现联网的话，得先搞清楚自己所处的网络环境。我的情况是有一段分配的IP，我等于是在服务器上用一个静态IP，容器上用另外一个静态IP，如果再添加容器就用第三个静态IP，（容器的静态IP是在每个容器内系统内设置的）。你要是也有这个条件的话，照我的方案应该也能用，关于其他情景下的网络配置，抱歉，实在是很懵逼，当时也折腾了很久网络的。
2. lxd的容器的安装位置能不能改变？不是很理解这个问题。如果是要迁移容器系统的话，就考虑备份啊镜像啊这种；如果你是觉得自己lxd配置的不对，分配的路径不合适，倒不如重新安装配置。
3. 对的，目前我就是在用分配的容器在使用，也给同学分配了容器，都能正常使用，而且是远程的SSH，如果你有遇到远程访问的困难，欢迎继续讨论，我也踩过很多坑。主要还是把容器上网的问题解决，剩下的都好说。

**Comment by fungtion** [comment]
好的，谢谢您的解答。关于第二个问题，我的疑问在于，容器本身是否是安装到了所设置的zfs存储池中？我看到显示的容器的位置是在/var中，是否是真正地装在了/var中，还是表面上安装在了/var中，zfs存储池挂载到了/var中呢？

**Comment by veraposeidon** [comment]
通过存储占用就可以看出来容器是装在哪的。
我的通过查看zpool list 就可以查看lxd所占用的大小了。

**Comment by fungtion** [comment]
好的，谢谢

**Comment by cfenglv** [comment]
你好，有一个问题想请教一下：
因为我们学校的网络需要有校园网登陆这个操作，所以我想的实现的是宿主机登陆了校园网，然后上面的虚拟机就能够上网。但是搭网桥的话是不是每个虚拟机都得单独登陆校园网呢？如果不搭网桥的坏处是什么呢？没有独立ip吗？

**Comment by veraposeidon** [comment]
你好，我对网桥这一块不是太清楚。
我认为网桥肯定是要搭的，不然容器没有通道能够上网了。
至于你需要的宿主机登陆校园网容器就能上网，应该还是属于网络配置里面的吧，我也不懂，冒汗。
还有一点，如果你实现了的话，容器校园网内是没有独立IP的，那就不容易实现局域网内的SSH登录了；不过，远程SSH登录倒是不受影响。
不好意西，没能帮到你。如果你有解决方案了，欢迎能够交流一波。

**Comment by cfenglv** [comment]
我现在的做法是，使用lxd init默认搭建的网桥，这样他会共享宿主机的网络，但是不会有独立外网ip。我用端口转发来解决这个问题，用户使用要通过ssh宿主机的某个端口来访问虚拟机。感觉这样对用户稍微麻烦一点，但是起码不用自己登陆校园网，但是还有一个问题就是宿主机的ip好像是动态的，我不知道怎么搞这个。。。要是宿主机ip变了，一段端口转发规则要改，感觉很烦。

**Comment by veraposeidon** [comment]
1. 动态IP是因为你的物理机使用的DHCP方式，然后校园网动态给你分配的IP，其实你可以尝试使用静态的方式给你的物理机固定个IP，就是拿你现在分配到的IP固定下来，但是有一定几率下次重启的时候IP跟别人的冲突了。
2. 还有就是你的服务器不是一直开着的吗，一直开着的话IP不变，这个问题也就不是问题。要不你试着写个脚本，开机检测IP，然后改写转发规则。
3. 还有一种方式就是SSH反向登录的方式，找个外网服务器做中转站，然后转发就好了（这种方式下，只要你的容器能上网就OK了）。但这个绝对是没有局域网内的速度和效率高的，实际上是远程办公的选择。你要尝试这种的话我可以再跟你细说

**Comment by cfenglv** [comment]
谢谢你给的方案。我试着用netplan固定ip之后，发现会变得很不稳定，可能是和别人冲突导致的，所以我现在就是尽量不关机。然后ssh反向登陆我其实没懂是什么意思，但是我现在其实也是由公网ip+固定端口号来远程访问虚拟机。（今天我们服务器的硬盘崩了一块，我所有lxd配置都在上面，还没来得及备份。。我现在得重新配置。。血一般的教训，看来zfs一定要加一块mirror。。。）
